pkg/kube/cluster-init.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/ash
+#
+# Copyright (c) 2023 Zededa, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+K3S_VERSION=v1.26.3+k3s1
+KUBEVIRT_VERSION=v0.59.0
+LONGHORN_VERSION=v1.4.2
+CDI_VERSION=v1.56.0
+
+INSTALL_LOG=/var/log/install.log
+date >> $INSTALL_LOG
+
+logmsg() {
+   local MSG
+   local TIME
+   MSG="$*"
+   TIME=$(date +"%F %T")
+   echo "$TIME : $MSG"  >> $INSTALL_LOG
+}
+
+setup_cgroup () {
+   echo "cgroup /sys/fs/cgroup cgroup defaults 0 0" >> /etc/fstab
+}
+
+
+                       
+check_network_connection () {
+ while true; do
+
+   IP="8.8.8.8"
```

Comment:
This might be installed in a network where the DNS server we get from DHCP is fine and we can reach https://get.k3s.io over port 443, but 8.8.8.8 and/or outbound DNS/UDP from the devices is blocked by the enterprise.

I think it would make more sense to check and wait until https://get.k3s.io can be reached before trying to fetch from there. 
But do we even need to check that once we have /var/lib/all_components_initialized?
In general EVE-OS should boot in disconnected mode so even a check for a default route is suspect.

Reply:
> This might be installed in a network where the DNS server we get from DHCP is fine and we can reach https://get.k3s.io over port 443, but 8.8.8.8 and/or outbound DNS/UDP from the devices is blocked by the enterprise.
> 
> I think it would make more sense to check and wait until https://get.k3s.io can be reached before trying to fetch from there. But do we even need to check that once we have /var/lib/all_components_initialized? In general EVE-OS should boot in disconnected mode so even a check for a default route is suspect.

Ok will check with @naiming-zededa on network part. But we need to check that even when we have all components installed because we cannot start k3s pods until network is available.

Reply:
maybe we can change to using 'curl -o /dev/null -w "%{http_code}" -s "https://get.k3s.io"' to see if it returns 200, instead of pinging 8.8.8.8; the '/var/lib/all_components_initialized' is done at the image upgrade, after that every reboots will need to check; In the current stage of this, k3s (in the container) actually checks for default route inside its code, we had an issue the k3s actually exited due to default route is not in the container side. So we are checking before launching the k3s server. In the later stage, I think we need to pass in the interface we want to have the k3s to use (instead of using a default route interface as today)

Reply:
The default route check is fine, but the ping one can be problematic.

Checking that https://get.k3s.io can get some response from the server makes sense. Detailed check might need some care though.